docs/contributing.md

```diff
@@ -0,0 +1,83 @@
+# Contributing
+
+This guide is for people who would like to be involved in building Protosearch.
+
+
+## Welcome!
+
+Firstly, welcome and thanks for your interest in Protosearch!
+
+
+## Getting Started
+
+We hope to maintain a list of [good first issues] to serve as starting points for new contributors.
+Occasionally this list might run empty, in which case, feel free to open a new issue asking for suggestions on where to get started.
+
+If you see an issue that interests you, and no one is currently assigned to it, please comment on the issue stating your interest.
+
+
+## Building The Project
+
+First you'll need to get the code, likely by cloning the repo locally:
+
+```sh
+git clone https://github.com/cozydev-pink/protosearch.git
+```
+
+Protosearch uses [sbt] and [sbt-typelevel] to manage its build.
+
+Here are some common tasks and their sbt commands:
+
+| Task             | sbt command          |
+| --------------   | -------------------- |
+| prepare for CI   | `prePR`              |
+| format code      | `scalafmtAll`        |
+| run JVM tests    | `rootJVM/test`       |
+| preview the site | `docs/tlSitePreview` |
+
+Protosearch cross compiles to multiple platforms, so commands that run for each platform, like `test` can take a while.
+It's often nice to focus on a single platform at first, and then test the others.
+For example, to run the tests in core for just Scala.js run `coreJS/test`.
+
+
+## Running The Project
+
+Currently the best way to run a modified version of protosearch is to publish the sbt plugin locally.
+We can do that in 3 steps:
+
+1. Publish locally with `++2.12.19 publishLocal`
+2. Add `protosearch-sbt` coords to `plugins.sbt`
+3. Run `tlSite`
+
+In step 1 we publish all modules for Scala 2.12 which is what all sbt plugins use.
+In step 2 we grab the version for our locally published protosearch from the log output of step 1.
+You'll likely see something like the following:
+
+```
+delivering :: pink.cozydev#protosearch-sbt;0.0-083dc6f-SNAPSHOT :: 0.0-083dc6f-SNAPSHOT ...
+```
+
+It's that `0.0-083dc6f-SNAPSHOT` bit that we want (Note that the version you see will be different).
+We can use that to add our locally published version of the `protosearch-sbt` plugin to another project's build.
+Let's add it to [http4s].
+
+Clone http4s, `git clone https://github.com/http4s/http4s.git`, and add the following to its `project/plugins.sbt` file:
+
+```
+addSbtPlugin("pink.cozydev" % "protosearch-sbt" % "0.0-083dc6f-SNAPSHOT")
+```
+
+Now we can run `tlSite` for the http4s docs to build an index using our locally published version of protosearch:
+
+```sh
+sbt site/tlSite
+```
+
+The above will produce `search.html` and `searchIndex.idx` files in the `site/target/docs/site/search/` directory.
+Starting a local web server and browsing to `search.html` should load the search.
```

Comment:
should we provide a pseudo-link here? In my head browsing == clicking, but there isn't a way to get to that page via clicks on the http4s site, folks would have to navigate manually. I'm worried that's not very clear but maybe it would be obvious... I'm not sure.

Reply:
You're exactly right.
I really flubbed this part because it really doesn't work the way I had hoped yet.
What I wanted to write was basically "run `site/tlSitePreview` and the go to this link in your browser"
But it turns out that the search index won't load in the `tlSitePreview` web server :(
Specifically the `searchIndex.idx` won't get served, it just 404s.

I don't know why yet.
I wonder if https://github.com/cozydev-pink/protosearch/pull/194 will change that...

Reply:
Unfortunately #194 won't change that. Something I was not aware of yet, is that the new `laikaRenderers` setting is not known by the preview server (which lives outside of the plugin project). So I'm afraid there is nothing you can do about this on your side, I will have to come up with an idea for this.

Reply:
I've created a follow up issue (https://github.com/cozydev-pink/protosearch/issues/199) to revisit this section and provide more clear instructions once we can leverage the preview server.